handlers/http.go

```diff
@@ -46,12 +51,17 @@ func (hrl httpRateLimiter) ServeHTTP(w http.ResponseWriter, r *http.Request) {
 	matches, err := hrl.rateLimiter.Add(request)
 	if err != nil && err != leakybucket.ErrorFull {
 		log.Printf("[%s] ERROR: %s", guid, err)
-		w.WriteHeader(500)
-		return
+		if !hrl.AllowOnError {
+			log.Printf("[%s] WARNING: bypassing rate limiter due to Error")
```

Comment:
Even if we do, we should probably metric it. Redis down !== Sphinx cannot contact redis.  (eg, network partition) .